nautobot/extras/forms/mixins.py

```diff
@@ -394,6 +400,33 @@ def clean(self):
           verify that the requested "source" object does not have an existing RelationshipAssociation to
           a different destination object.
         """
+
+        required_relationships = Relationship.objects.get_required_for_model(self.instance)
+        for side, relations in required_relationships.items():
+            for relation in relations:
+
+                if relation.symmetric:
+                    if not self.cleaned_data.get(f"cr_{relation.slug}__peer", []):
+                        model_name = relation.source_type.model_class()._meta.verbose_name
+                        # Only add the error once for this field
+                        if len(self.errors.get(f"cr_{relation.slug}__peer", [])) == 0:
+                            self.add_error(
+                                f"cr_{relation.slug}__peer", forms.ValidationError(f"You must select a {model_name}")
+                            )
+                    continue
+
+                # Skip self-referencing
+                if ContentType.objects.get_for_model(self.instance) == getattr(relation, f"{relation.required}_type"):
+                    continue
+
+                model_name = getattr(relation, f"{side}_type").model_class()._meta.verbose_name
+                cr_field_name = f"cr_{relation.slug}__{relation.required}"
+
+                if not self.cleaned_data[cr_field_name]:
+                    self.add_error(
+                        f"cr_{relation.slug}__{side}", forms.ValidationError(f"You must select a {model_name}")
```

Comment:
Need to think about how to make this message accurate for both `*-to-1` relations and `*-to-many` relations.